langchain_dappier/retrievers.py

```
@@ -0,0 +1,212 @@
+                    query=query,
+                    data_model_id=self.data_model_id,
+                    similarity_top_k=kwargs.get("k", self.k),
+                    ref=self.ref,
+                    num_articles_ref=self.num_articles_ref,
+                    search_algorithm=self.search_algorithm,
+                )
+                return self._extract_documents(response=response)
+        except Exception as e:
+            raise ValueError(f"Error while retrieving documents: {e}") from e
+
+    def _extract_documents(self, response: Any) -> List[Document]:
```

Ruff (0.8.2)

Dynamically typed expressions (typing.Any) are disallowed in `response`

(ANN401)